src/http/http_client.f90

```diff
@@ -142,6 +153,33 @@ & function failed. This can occur due to insufficient memory available in the sy
         call curl_easy_cleanup(curl_ptr)
       
     end function client_get_response
+    
+    ! This function converts the request%form data into URL-encoded name-value pairs
+    ! and stores the result in the request%form_encoded_str variable. The resulting 
+    ! string is used in an HTTP request body with the application/x-www-form-urlencoded
+    !  content type to send data as name-value pairs.
+    subroutine prepare_form_encoded_str(curl_ptr, request) 
+        type(c_ptr), intent(out) :: curl_ptr
+        type(request_type), intent(inout) :: request
```

Comment:
See [http_header.f90](https://github.com/fortran-lang/http-client/blob/main/src/http/http_header.f90) for example of how to write FORD-style docstrings.